scipy/stats/_grenander.py

```diff
@@ -0,0 +1,260 @@
+
+    .. [2] U. Grenander, "On the theory of mortality measurement: part II", 
+           Scandinavian Actuarial Journal, Vol. 39, pp. 125-153, 1956. 
+           
+    .. [3] P. Groeneboom and G. Jongbloed, "Nonparametric estimation under 
+           shape constraints", Cambridge University Press, New York, 2014. 
+           
+    .. [4] B.L.S.P. Rao, "Estimation of a unimodal density", Sankhyā: The 
+           Indian Journal of Statistics, Series A, Vol. 31, pp. 23-36, 1969.
+
+    Examples
+    --------
+    Estimate a decreasing density from exponential data.
+
+    >>> import numpy as np
+    >>> from scipy.stats import expon, grenander
+    >>> rng = np.random.default_rng(1234567891)
+    >>> x = rng.exponential(size=100)
+    >>> g = grenander(x)
+
+    The estimator provides the fitted density values at the observations:
+
+    >>> g.fitted.shape
+    (100,)
+
+    Evaluate the PDF and CDF at specific points:
+
+    >>> g.pdf([0.5, 1.0, 2.0])
+    array([0.52828842, 0.23170026, 0.21893189])
+    >>> g.cdf([0.5, 1.0, 2.0])
+    array([0.37589365, 0.60242482, 0.83153771])
+
+    The estimator is piecewise constant with data-adaptive breakpoints:
+
+    >>> len(g.slopes)  # number of constant pieces
+    15
+    >>> g.knots  # breakpoints of the histogram
+    array([0.        , 0.00903572, 0.02683709, 0.05754583, 0.46991483,
+           0.73492159, 0.77473536, 0.98953465, 1.11901228, 1.9016237 ,
+           2.08432892, 2.29638573, 2.68628217, 3.38105853, 4.67484783,
+           5.93619219])
+
+    Draw new samples from the estimated density:
+
+    >>> y = g.rvs(size=1000, random_state=0)
+    >>> y.shape
+    (1000,)
+
+    Visualize the estimated density alongside the true exponential density
+    and the empirical CDF:
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 4))
+
+    >>> # Plot density
+    >>> xs = np.linspace(0, 6, 500)
+    >>> ax1.plot(xs, expon.pdf(xs), 'k-', label='True density')
+    >>> g.plot_pdf(ax=ax1, fill=True, alpha=0.7, label='Grenander')
+    >>> ax1.set_xlabel('x')
+    >>> ax1.set_ylabel('Density')
+    >>> ax1.legend()
+
+    >>> # Plot CDF
+    >>> ax2.plot(xs, g.cdf(xs), label='Grenander CDF')
+    >>> ax2.step(np.sort(x), np.arange(1, len(x)+1)/len(x), 
+    ...          where='post', label='Empirical CDF')
+    >>> ax2.plot(xs, expon.cdf(xs), 'k-', label='True CDF')
+    >>> ax2.set_xlabel('x')
+    >>> ax2.set_ylabel('CDF')
+    >>> ax2.legend()
+    >>> plt.tight_layout()
+    >>> plt.show()
+    """
+    def __init__(self, 
+                 dataset, 
+                 support_min=0.0, 
+                 assume_sorted=False):
+        self.support_min = support_min
+        params = self._fit(dataset, assume_sorted)
+        self.__dict__.update(params)
+
+        self._cdf_interp = interp1d(
+            self.knots, self.heights, kind="linear",
+            bounds_error=False, fill_value=(0.0, 1.0),
+            assume_sorted=True
+        )
```

Comment:
It would be good to define the inverse CDF by reversing the order of the `self.heights` and `self.knots` arguments. Then, if you use the new distribution infrastructure, you'd also get the inverse distribution function methods and a fairly efficient `sample` (replaces `rvs`) method for free.